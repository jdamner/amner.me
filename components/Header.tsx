import React from "react";

export default function Header(
  props: React.DetailedHTMLProps<
    React.HTMLProps<HTMLDivElement>,
    HTMLDivElement
  > & { title: string },
) {
  const { title, children, ...rest } = props;

  return (
    <section
      id="header"
      className="bg-black corner-blob text-orange-100 py-4 px-4"
      {...rest}
    >
      <div className="container py-5 mx-auto ">
        <h1 className="font-sans text-5xl sm:text-8xl lg:text-9xl font-black uppercase mb-5 lg:w-1/2">
          {title}
        </h1>
        <div className="text-xl md:text-2xl prose max-w-screen-md">
          {children}
        </div>
      </div>
    </section>
  );
}
