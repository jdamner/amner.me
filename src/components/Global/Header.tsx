/* Types */
import type { withChildren } from "../../types/children.type";

/**
 * Header 
 * @param {JSX.IntrinsicElements['section'] & withChildren & { subtitle?: string } } props
 * @returns {JSX.Element}
 */
export default function Header(
    { subtitle, title, children } : JSX.IntrinsicElements['section'] & withChildren & { subtitle?: string }
): JSX.Element {

    const subtitleClasses = [
        'font-sans',
        'text-lg',
        'md:text-xl',
        'font-black',
        'uppercase',
        'flex',
        'justify-center',
        'items-center',
        'w-full',
    ];

    const subtitleSpanClasses = [
        'inline-block',
        'bg-slate-900',
        'dark:bg-slate-200',
        'h-2',
        'last:rotate-180',
        'grow',
        'hidden',
        'md:block',
        'w-1',
        'md:w-full',
    ]

    return (
        <section id="header" className='bg-slate-200 dark:bg-slate-800 text-slate-900 dark:text-slate-300 py-5'>
            <div className='container mx-auto text-center py-5 px-3 md:px-0'>
                {subtitle && <h2 className={subtitleClasses.join(' ')}>
                    <span className={subtitleSpanClasses.join(' ')}></span>
                    <span className='my-3 md:basis-1/2'>{subtitle}</span>
                    <span className={subtitleSpanClasses.join(' ')}></span>
                </h2>}
                <h1 className="font-sans text-4xl md:text-9xl font-black uppercase mb-5">{title}</h1>
                {children}
            </div>
        </section>
    );
}
