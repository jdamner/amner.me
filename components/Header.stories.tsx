import type { Meta, StoryObj } from "@storybook/nextjs";

import Header from "./Header";

type Story = StoryObj<typeof Header>;

export default {
  title: "Components/Header",
  component: Header,
} as Meta;

export const Default: Story = {
  args: {
    title: "Hi, I'm a web developer.",
    children: "I build accessible, high performance web applications.",
  },
};
