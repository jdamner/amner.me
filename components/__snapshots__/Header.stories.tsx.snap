// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Header Default smoke-test 1`] = `
<div class="bg-black text-orange-100  font-sans">
  <section id="header"
           class="bg-black corner-blob text-orange-100 py-4 px-4"
  >
    <div class="container py-5 mx-auto ">
      <h1 class="font-sans text-5xl sm:text-8xl lg:text-9xl font-black uppercase mb-5 lg:w-1/2">
        Hi, I'm a web developer.
      </h1>
      <div class="text-xl md:text-2xl prose max-w-screen-md">
        I build accessible, high performance web applications.
      </div>
    </div>
  </section>
</div>
`;
